casper/src/test/scala/coop/rchain/casper/MergeabilityRules.scala

```diff
@@ -328,6 +341,39 @@ trait MergeabilityRules {
 
   }
 
+  /***
+   There was a COMM within one of the deploys caused on one side by a peek.
+   The other deploy had an event without a match in TS, of same polarity to the data matched by peek.
+   These could not spawn more work.
+    */
+  object PeekedNoMatch extends MergeableCase {
+
+    override def precondition(
+        left: Rho*
+    )(right: Rho*)(base: Rho*)(implicit pos: source.Position): Effect[_] =
+      volatileEventPrecondition(left: _*)(right: _*)(base: _*) >> {
+        assert(atLeastOnePeek(left ++ right))
+      }.pure[Effect]
+
+  }
+
+  /***
+   There was a COMM within one of the deploys, with a peek on one side.
+   The other deploy had an event without a match in TS, dual to the non-linear.
+   These could spawn more work.
+   Mergeable if we use spatial matcher to prove they don't match.
+    */
+  object PeekedCouldMatch extends ConflictingCase {
+
+    override def precondition(
+        left: Rho*
+    )(right: Rho*)(base: Rho*)(implicit pos: source.Position): Effect[_] =
+      volatileEventPrecondition(left: _*)(right: _*)(base: _*) >> {
+        assert(atLeastOnePeek(left ++ right))
```

Comment:
We should differentiate the precondtions between the `Peeked*` cases. Or, ideally, we'd incorporate peek cases into the existing ones.

Reply:
The latter is not possible. I already promised Dom I'm going to improve the preconditions https://github.com/rchain/rchain/pull/2692#discussion_r323189222